docs/resource-ids.md

```diff
@@ -0,0 +1,51 @@
+# Resource IDs
+
+Every Pulumi resource must have an `"id"` field. `"id"` must be a string, and it must be
+set by the provider (`Computed` and *not* `Optional` in Terraform parlance).
+
+## SDKv1 and SDKv2 Based Providers
+
+The ID requirement is easy to satisfy for SDKv{1,2} based providers since both SDKs
+require an ID field of type string set on the provider. If the provider being bridged is
+based on SDKv1 or SDKv2, then the bridge handles Pulumi's ID field without intervention.
+
+## PF Based Providers
+
+Most PF based providers have an attribute called `"id"` of the right kind for the bridge
+to use. If your provider doesn't, then you will see an error when `make tfgen` is
+run. Each error has a different kind of resolution.
+
+### ID of the wrong type
```

Comment:
Section on missing "id" fields? I was recently populating ComputeID in aws again. Did we deprecate that or that's still the way?